static/js/components/PersonalTab.js

```diff
@@ -68,7 +68,15 @@ export default class PersonalTab extends React.Component {
     dispatch(setProfileStep(PERSONAL_STEP));
   }
 
-  onChange = (selection: Option): void => {
+  componentDidUpdate() {
+    const { currentProgramEnrollment } = this.props;
+    const selectedProgram = this.getSelectedProgramId();
+    if ( currentProgramEnrollment && !selectedProgram ) {
+      this.onProgramSelect({ value: currentProgramEnrollment.id, label: '' });
```

Comment:
Oh, sorry, I went with the original suggestion, the `componentDidUpdate` function just pulls `setProgram` out of `this.props` and calls it with `currentProgramEnrollment` as an argument.